x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/operators/convert_inference_events_to_streaming_events.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Observable, OperatorFunction, map } from 'rxjs';
+import { v4 } from 'uuid';
+import {
+  ChatCompletionEvent as InferenceChatCompletionEvent,
+  ChatCompletionEventType as InferenceChatCompletionEventType,
+} from '@kbn/inference-common';
+import {
+  ChatCompletionChunkEvent,
+  TokenCountEvent,
+  ChatCompletionMessageEvent,
+  StreamingChatResponseEventType,
+} from '../../../../common';
+
+export function convertInferenceEventsToStreamingEvents(): OperatorFunction<
+  InferenceChatCompletionEvent,
+  ChatCompletionChunkEvent | TokenCountEvent | ChatCompletionMessageEvent
+> {
+  return (events$: Observable<InferenceChatCompletionEvent>) => {
+    return events$.pipe(
+      map((event) => {
+        if (event.type === InferenceChatCompletionEventType.ChatCompletionChunk) {
+          // Convert to ChatCompletionChunkEvent
+          return {
+            type: StreamingChatResponseEventType.ChatCompletionChunk,
+            id: v4(),
+            message: {
+              content: event.content,
+              function_call:
+                event.tool_calls.length > 0
+                  ? {
+                      name: event.tool_calls[0].function?.name,
+                      arguments: event.tool_calls[0].function?.arguments,
+                    }
```

Comment:
NIT: `ChatCompletionChunkToolCall.function.name/arguments` isn't optional, so I think you can drop the `?` operator here.

https://github.com/elastic/kibana/blob/fe168221df0d0bf598a8c32eb3c910df402572db/x-pack/packages/ai-infra/inference-common/src/chat_complete/events.ts#L51-L60

Reply:
done